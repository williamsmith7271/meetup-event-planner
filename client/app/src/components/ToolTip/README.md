## ToolTip Component
A component that ...

### Example

```js
<ToolTip />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
