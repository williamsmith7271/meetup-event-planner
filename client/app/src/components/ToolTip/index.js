import React, { PropTypes } from 'react';
import Tip from 'grommet-udacity/components/Tip';

const ToolTip = ({
  isShowing,
  children,
  onClose,
}) => (
  <div>
    {isShowing &&
      <Tip size="medium" target="passwordInput" onClose={onClose}>
        {children}
      </Tip>
    }
  </div>
);

ToolTip.propTypes = {
  isShowing: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default ToolTip;
