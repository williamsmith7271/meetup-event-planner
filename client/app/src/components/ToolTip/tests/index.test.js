import ToolTip from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<ToolTip />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <ToolTip />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
