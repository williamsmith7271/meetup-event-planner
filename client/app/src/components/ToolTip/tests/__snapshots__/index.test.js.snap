exports[`<ToolTip /> should render with default props 1`] = `<div />`;
